test/jdk/tools/launcher/InstanceMainTest.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+/**
+ * @test
+ * @summary test execution priority of main methods
+ * @run main InstanceMainTest
+ */
+public class InstanceMainTest extends TestHelper {
```

Comment:
By my reading of the spec, "main" methods can be defined in record classes and enum classes as well as normal classes.
Are there tests for record and enum main method operation?

Reply:
You can also have a `main` method inside an interface. For enum classes and interfaces, the main has to be static.

Reply:
updating tests